glossary.md

```diff
@@ -0,0 +1,89 @@
+# Glossary of Optimistic Ethereum Terms
+
+### Layer 1 (L1)
+
+The Ethereum Mainnet.
+
+### Layer 2 (L2)
+
+The Optimistic Ethereum Network.
+
+### State Root Oracle
+
+A smart contract on L1, providing access to historical L1 block hashes. The existence of this contract makes it possible to prove the existence of a particular L2 state root within a previous L1 block.
+
+### Block Generation (AKA Block Derivation)
+
+All of the data necessary to reconstruct the history and state of L2 is contained within L1 blocks. Block Generation is the process of converting data contained in L1 blocks into L2 blocks.
+
+### Epoch
+
+A continuous sequence of 0 or more L2 blocks. There is a one to one correspondence between L2 Epochs, and L1 blocks.
+
+**Example usage:** "Epoch 20 corresponds to L1 block 637, epoch 21 corresponds to block 638, etc".
+
+### Feeds
+
+Feeds are append only lists, each of which is contained in a smart contract on Layer 1. The state of the Rollup Chain is determined by the data written to these contracts.
+
+### Sequencer Feed
+
+The Sequencer Feed is a contract which contains the data representing sequencer submitted blocks, each element in this feed will be used to generate a "sequencer block".
+
+### Deposit Feed
+
+The Deposit Feed is a contract which contains deposit data. Each element in this feed will be used to generate a "deposit block".
+
+### Sequencing Window
+
+The number of L1 blocks during which the Sequencer _may_ insert Sequencer blocks before a Deposit block.
+
+If the Sequencer Window is 2 blocks, and Alice makes a Deposit in Block 420, her deposit WILL be included (by the Block Generation process) at block 422.
+
+### Execution Environment (EE)
+
+The virtual machine in which transactions are applied to the pre-state to generate the post-state.
+
+### Rollup Node AKA Rollup Driver
+
+The core client software which reads data from an L1 node, and transforms it into L2 data. Both the Sequencer, and Verifier nodes run a Rollup Node, albeit in different configurations.
+
+### Consensus Layer
+
+By analogy to the architecture of ETH2 we can also describe Layer 1 as the "Consensus Layer", as it defines the canonical history and state of the Layer 2 chain.
+
```

Comment:
Do we want to add these analogies / overloaded terms? I think if we do this it becomes confusing because suddenly "consensus layer" can mean either the L1 consensus layer or the whole of L1.